### PR TITLE
Add documentation for tracking

### PR DESCRIPTION
**Pull request by vaidehijoshi**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [x] Documentation Update

## Description

Now that we have `ahoy_matey` and `ahoy.js` set up for first-party tracking, we need some documentation around these libraries, how to use them, and what configurations we have for them and why.

This PR adds all of that :)

I also renamed the "Tracking Metrics" documentation file to just be "Metrics", so that if someone greps our docs for `tracking`, that file won't show up.

## Related Tickets & Documents
N/A

## QA Instructions, Screenshots, Recordings

Check out the new documentation files, make sure they make sense/have no typos, and that they can link between each other!

## Added tests?

- [ ] yes
- [ ] no, because they aren't needed
- [x] no, because I need help

## Added to documentation?

- [x] docs.dev.to
- [ ] readme
- [ ] no documentation needed

## Are there any post deployment tasks we need to perform?
N/A

## What gif best describes this PR or how it makes you feel?

tracking gif

### CODE REVIEW

File: docs/backend/tracking.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+Ahoy creates an `Ahoy::Event` record for each event that it tracks. If no visit
+is recorded for a user when an event is tracked, Ahoy will simeltaneously create
+an `Ahoy::Visit` for the event being tracked.
+
+Events can be tracked in a controller action on the backend, or with JavaScript
+on the frontend. Learn more about tracking events with JavaScript in our
+[frontend tracking guide](../frontend/tracking.md).
+
+When an event is tracked, it should include a `name` and a `properties` hash.
+When adding new events, be sure that the name is unique per-event, and that the
```
Comment (Ridhwana) [inline docs/backend/tracking.md:46]: I find that the sentence reads a bit odd:
```
When adding new events, be sure that the name is unique per-event, and that the
properties will help you differentiate the different events with enough
granularity.
```
Should this be:
```
When adding new events, be sure that the name is unique per-event. The
properties will help you differentiate the different events with enough
granularity.
```

Comment (vaidehijoshi) [reply docs/backend/tracking.md:46]: Oh, I like that better. Updated!

File: docs/backend/tracking.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+## Visits & Events
+
+For first-party analytics, we use the
+[`ahoy_matey` gem](https://github.com/ankane/ahoy), which tracks visits and
+events.
+
+We intentionally choose to limit what user data we track and persist, and have
+opted to follow the GDPR compliance standards
+[set by Ahoy](https://github.com/ankane/ahoy#gdpr-compliance-1). By default, we
+have configured the Ahoy library to mask IP addresses, have disabled geocode
```
Comment (mstruve) [inline docs/backend/tracking.md:14]: "mask ip addresses, disable geocode tracking, and not track user cookies."

File: docs/backend/tracking.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+is recorded for a user when an event is tracked, Ahoy will simeltaneously create
+an `Ahoy::Visit` for the event being tracked.
+
+Events can be tracked in a controller action on the backend, or with JavaScript
+on the frontend. Learn more about tracking events with JavaScript in our
+[frontend tracking guide](../frontend/tracking.md).
+
+When an event is tracked, it should include a `name` and a `properties` hash.
+When adding new events, be sure that the name is unique per-event. The
+properties will help you differentiate between events with enough granularity.
```
Comment (mstruve) [inline docs/backend/tracking.md:47]: ~with enough granularity~

File: docs/backend/tracking.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+necessary, but this should be done so _with explict care_.
+
+We do not collect any personal user data when tracking visits. Our collected
+data is limited to the user's `id`. Each user has a unique `visitor_token`,
+while each visit to the site is marked with a unique `visit_token`.
+
+### Events
+
+Ahoy creates an `Ahoy::Event` record for each event that it tracks. If no visit
+is recorded for a user when an event is tracked, Ahoy will simeltaneously create
```
Comment (jacobherrington) [inline docs/backend/tracking.md:38]: simultaneously

Comment (vaidehijoshi) [reply docs/backend/tracking.md]: lol WHOOPS

**Pull request merged**